src/main/java/teammates/common/util/ActivityLogEntry.java

```diff
@@ -470,6 +470,33 @@ public static String generateSystemErrorReportLogMessage(HttpServletRequest req,
         
         return emailReportLog.generateLogMessage();
     }
+
+    public static String generateSystemErrorReportLogMessage(HttpServletRequest req, Sendgrid errorEmail) {
+        String[] actionTaken = req.getServletPath().split("/");
+        String action = req.getServletPath();
+        if(actionTaken.length > 0) {
+            action = actionTaken[actionTaken.length-1]; //retrieve last segment in path
+        }
+        String url = HttpRequestHelper.getRequestedURL(req);
+        
+        String message = "";
+        if(errorEmail != null){
+            try {
+                  message += "<span class=\"text-danger\">" + errorEmail.getSubject() + "</span><br>";
+                  message += "<a href=\"#\" onclick=\"showHideErrorMessage('error" + errorEmail.hashCode() +"');\">Show/Hide Details >></a>";
+                  message += "<br>";
+                  message += "<span id=\"error" + errorEmail.hashCode() + "\" style=\"display: none;\">";
+                  message += errorEmail.getHtml();
+                  message += "</span>";
+              } catch (Exception e) {
+                  message = "System Error. Unable to retrieve Email Report";
+              }
+          }
+        
+        ActivityLogEntry emailReportLog = new ActivityLogEntry(action, Const.ACTION_RESULT_SYSTEM_ERROR_REPORT, null, message, url);
+        
+        return emailReportLog.generateLogMessage();
+    }
```

Comment:
This is an overloaded method which takes Sendgrid as a parameter instead of MimeMessage.